frontend/packages/integration-tests-cypress/tests/app/demo-dynamic-plugin.spec.ts

```diff
@@ -0,0 +1,198 @@
+import { isLocalDevEnvironment } from '../../views/common';
+import { detailsPage } from '../../views/details-page';
+import { infoMessage, hintMessage, warningMessage, warningToast } from '../../views/form';
+import { listPage } from '../../views/list-page';
+import { modal } from '../../views/modal';
+import { nav } from '../../views/nav';
+
+const PLUGIN_NAME = 'console-demo-plugin';
+const PLUGIN_PATH = '../../../dynamic-demo-plugin';
+const CHECK_UPDATE_WAIT = 30000;
+const PLUGIN_PULL_SPEC = Cypress.env('PLUGIN_PULL_SPEC');
+
+/*
+  These tests are meant to:
+    1. show how to test a dynamic plugin using demo as the plugin instance
+    2. run locally:
+      2a. build the plugin locally, and run the server
+      2b. using bridge running with the plugin argument that points to the local dynanmic plugin server
+      2c. will intercept the check update call to mock a toast notification that a plugin has been enabled
+      2d. will not use all workload definitions defined in the yaml (not using the env variable for pull spec)
+    3. run on ci:
+      3a. ci will build the dynamic plugin and provide the pullspec in the env var: CYPRESS_PLUGIN_PULL_SPEC
+      3b. that pull spec will be used to create the deployment on the cluster
+      3c. intercepting the check-updates call will not occur as the toast will appear when the plugin is enabled
+    4. the scaffolding should remain the same except modifying the constants above
+ */
+
+const enableDemoPlugin = (enable: boolean) => {
+  // find console demo plugin and enable it
+  cy.visit('k8s/cluster/operator.openshift.io~v1~Console/cluster/console-plugins');
+  cy.url().should(
+    'include',
+    'k8s/cluster/operator.openshift.io~v1~Console/cluster/console-plugins',
+  );
+  cy.get('.co-resource-item__resource-name')
+    .byLegacyTestID(PLUGIN_NAME)
+    .should('be.visible');
+  cy.byLegacyTestID(PLUGIN_NAME)
+    .parents('tr')
+    .within(() => {
+      cy.byTestID('edit-console-plugin').contains(enable ? 'Disabled' : 'Enabled');
+      cy.byTestID('edit-console-plugin').click();
+    });
+  modal.shouldBeOpened();
+  cy.contains('Cancel');
+  modal.modalTitleShouldContain('Console plugin enablement');
+  cy.byTestID(enable ? 'Enable-radio-input' : 'Disable-radio-input').click();
+  modal.submit();
+  modal.shouldBeClosed();
+  cy.byLegacyTestID(PLUGIN_NAME)
+    .parents('tr')
+    .within(() => {
+      cy.byTestID('edit-console-plugin').contains(enable ? 'Enabled' : 'Disabled');
+    });
+  if (isLocalDevEnvironment) {
+    // for local dev env just trigger any change in the return value to activate the toast
+    cy.intercept('/api/check-updates*', { plugins: [] }).as('checkUpdates');
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(CHECK_UPDATE_WAIT);
+    cy.wait('@checkUpdates');
+    cy.log('Running plugin test locally using bridge.');
+  } else {
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(CHECK_UPDATE_WAIT);
```

Comment:
I'm not sure if hard wait is the best solution

Reply:
Hard wait is needed because there is poll interval that we need to accommodate during testing. The toast is not triggered when the change occurs, rather it is triggered after polling the system twice and comparing the plugin values. Thats why a wait is needed in this case @yapei